### PR TITLE
Playlist feature

### PR DESCRIPTION
**Pull request by Mohitkumar33**
Added the playlist functionality
Add to playlist
delete playlist
remove playlist

### CODE REVIEW

File: src/contexts/playlistContext/playlist-context.js [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+
 const playlistContext = createContext(null);
 
 const PlaylistProvider = ({ children }) => {
+  const { authState } = useAuth();
+  const { isAuth } = authState;
   const [playlistState, playlistDispatch] = useReducer(playlistReducer, {
     playlists: [],
   });
+  useEffect(() => {
```
Comment (deepak29-git) [inline src/contexts/playlistContext/playlist-context.js:14]: you can use different function for api call

File: src/contexts/playlistContext/playlistReducer.js [hunk 2]
```diff
@@ -1,5 +1,26 @@
 const playlistReducer = (playlistState, playlistAction) => {
```
Comment (deepak29-git) [inline src/contexts/playlistContext/playlistReducer.js:1]: this is good to keep reducer in separate file

File: src/utilities/playlistsUtils.js [hunk 3]
```diff
@@ -1,4 +1,120 @@
...
+  }
+};
+
+const deleteVideoInPlaylist = async (playlistId, videoId, playlistDispatch) => {
+  try {
+    const { data } = await axios.delete(
+      `${playlist}/${playlistId}/${videoId}`,
+      {
+        headers: {
+          authorization: localStorage.getItem("token"),
```
Comment (deepak29-git) [inline src/utilities/playlistsUtils.js:101]: you can make a function for token and use it everywhere

**Review by deepak29-git — APPROVED** [review]

File: src/contexts/playlistContext/playlistReducer.js (hunk 2 above)
Comment (Mohitkumar33) [reply src/contexts/playlistContext/playlistReducer.js:1]: Thanks

File: src/pages/playlist/playlist.js [hunk 4]
```diff
@@ -1,9 +1,51 @@
...
   return (
     <>
-      <p>This is playlist</p>
+      <main className="main-content">
+        <div className="playlist-heading">
+          <h1>My playlist</h1>
+          {/* <button>Create New Playlist</button> */}
+        </div>
+        <div className="all-playlist">
+          {playlists &&
```
Comment (SukanyaSen530) [inline src/pages/playlist/playlist.js:17]: You can replace the additional check with (playlist?.map)

File: src/utilities/playlistsUtils.js (hunk 3 above)
Comment (Mohitkumar33) [reply src/utilities/playlistsUtils.js:101]: But I will have to import it everywhere and also write that in front of authorization. Thats why I wrote like this only

File: src/pages/playlist/playlist.js (hunk 4 above)
Comment (Mohitkumar33) [reply src/pages/playlist/playlist.js:17]: Ok

**Review by SukanyaSen530 — APPROVED** [review]
Suggested a small change. Rest look's good.

**Pull request merged**